### ISSUE TITLE
wrong Content-Length if umlauts

### ISSUE THREAD

**Issue by hillar** [lucasdiedrich/node-freeipa#23]
if one sends some umlauts, like

```
{
  reqOpts: {
    method: 'POST',
    host: 'ipa.demo1.freeipa.org',
    path: '/ipa/session/json',
    ca: false,
    headers: {
      accept: 'application/json',
      'content-type': 'application/json; charset=utf-8',
      referer: 'https://ipa.demo1.freeipa.org/ipa',
      'Content-Length': 102,
      Cookie: 'ipa_session=MagBearerToken=EwJUHVmjhsE%2bgdYynVpsZxjuRNANRzaLO0yerNt%2bREudrdA1LFTI7dVWf4KYfwxVkCqYGJS0iWrcg3vicmIkm3bewpWt1JnpEaVTwTNY3gS8hvoMEmw4qpk62Fttft7zKm6UD6x2Y93Sf%2f7wlf34fb6%2b75yeAruVgblzrE1aNG6mw1sfvuQ%2fdscoeld2WFxAqCifJyUpGyqtXrG2M8qiHQ%3d%3d;path=/ipa;httponly;secure;'
    },
    rejectUnauthorized: false
  },
  data: '{"method":"stageuser_add","params":[["test-rpc"],{"givenname":"stage","sn":"öäüõ","version":"2.164"}]}'
}
```

result will be:

```

{
  code: 909,
  message: 'Invalid JSON-RPC request: Unterminated string starting at: line 1 column 93 (char 92)',
  data: {
    error: 'Unterminated string starting at: line 1 column 93 (char 92)'
  },
  name: 'JSONError'
}

```

as content-length does not match real length and input will be chopped ;(

https://github.com/freeipa/freeipa/blob/c053b5e0d6a06ad17b1d427a8345bca792981b47/ipaserver/rpcserver.py#L228

```
    try:
        length = int(environ.get('CONTENT_LENGTH'))
    except (ValueError, TypeError):
        return None
    return environ['wsgi.input'].read(length).decode('utf-8')
```

i think fix will be to replace

ref https://github.com/lucasdiedrich/node-freeipa/blob/a089c8e346d30e5db267c8e2a645dbf155fdf27d/src/lib/RequestBuilder.js#L104
```
'Content-Length': data.length,
```
with

```
'Content-Length': Buffer.byteLength(data),
```

**Comment by lucasdiedrich** [comment]
@hillar have you tested your PR? it fixed your problem?

**Comment by hillar** [comment]
yes

**Comment by lucasdiedrich** [comment]
Thank you @hillar , its has been published as node-freeipa@2.3.2
